contracts/supertoken/interfaces/IHook.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity ^0.8.3;
+
+interface IHook {
+    /// @dev This function calls the srcHookCall function of the connector contract,
+    /// passing in the receiver, amount, connector, and msg.sender, and returns
+    /// the updated receiver, amount, and extradata.
+    /// @param receiver_ The receiver of the funds.
+    /// @param amount_ The amount of funds.
+    /// @param siblingChainSlug_ The address of the connector contract.
+    /// @param connector_ The address of the connector contract.
+    /// @param msgSender_ The address of the message sender.
+    /// @param extradata_ Additional data to be passed to the connector contract.
+    /// @return receiver The updated receiver of the funds.
+    /// @return amount The updated amount of funds.
+    /// @return extradata The updated extradata.
+    function srcHookCall(
```

Comment:
can we call this validateAndUpdateInputs() or something like that?

can we enforce this on the interface level so it doesnt do any state-updates? make it view or something and do a staticcall to the hook

Reply:
I think we wont be able to do that, in case of yield hook we would have to call dl and have state changes there

Reply:
even in Limit hook we have consumeFullLimit, which does state updates